### PR TITLE
document the sync with linguist

### PR DESCRIPTION
**Pull request by mcarmonaa**
document how and when we sync with linguist

### CODE REVIEW

**Comment by bzz** [comment]
Looks good to me

File: README.md [hunk 1]
```diff
@@ -35,9 +35,15 @@ lang := enry.GetLanguage("foo.cpp", "<cpp-code>")
 
 Developmemt
 -----------
-*enry* re-uses parts of original [linguist](https://github.com/github/linguist) especially data in `languages.yml` to generate internal data structures. In oreder to update to latest upstream run
+*enry* re-uses parts of original [linguist](https://github.com/github/linguist)  to generate internal data structures. In order to update to latest upstream and generate the necessary code you must run:
 
-    make clean code-generate
+    go generate
+
+We update enry due to changes  in linguist's master branch related to the following files:
```
Comment (mcuadros) [inline README.md]: When is this changes happening?

Comment (mcarmonaa) [inline README.md]: We don't have a procedure to detect automatically changes in linguist and sync in fact, it happens when during coding some new functionality of fixing something we run go generate.

Comment (mcuadros) [inline README.md]: But when are you updating this? After witch actions? Randomly, if yes document it. The people can open a request with the update? Or they can request the update?

Comment (mcarmonaa) [inline README.md]: Ok, do you prefer some specific policy for this?

Comment (smola) [inline README.md]: I would say that we say we do this periodically and that people can open both issues requesting it or PRs doing it.

Comment (mcarmonaa) [inline README.md]: It's commented in this way now,

* For the moment we don't have any procedure established to detect changes in the linguist project automatically and regenerate the code. So we are updating the generated code when is needed  without any specific criteria while we are going ahead in the project development.

 * If you want update *enry* because of changes in linguist, you can run the *go generate* command and do a pull request that only contains the changes in generated files (those files in the subdirectory [data](https://github.com/src-d/enry/tree/master/data)).

File: README.md [hunk 2]
```diff
@@ -35,9 +35,19 @@ lang := enry.GetLanguage("foo.cpp", "<cpp-code>")
 
 Developmemt
 -----------
-*enry* re-uses parts of original [linguist](https://github.com/github/linguist) especially data in `languages.yml` to generate internal data structures. In oreder to update to latest upstream run
+*enry* re-uses parts of original [linguist](https://github.com/github/linguist)  to generate internal data structures. In order to update to latest upstream and generate the necessary code you must run:
 
-    make clean code-generate
+    go generate
```
Comment (mcuadros) [inline README.md]: Why this change? The male is required to download the original linguist, isn't is?

Comment (mcarmonaa) [inline README.md]: `go generate` runs `make code-generate` that download linguist if it's not downloaded, do we want to run the `clean` rule too and always download linguist?

**Comment by abeaumont** [comment]
@mcarmonaa please rebase once review is finished.

File: README.md [hunk 3]
```diff
@@ -115,11 +116,21 @@ Note that even if enry's CLI is compatible with linguist's, its main point is th
...
+
+We update enry due to changes  in linguist's master branch related to the following files:
+* [languages.yml](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml)
+* [heuristics.rb](https://github.com/github/linguist/blob/master/lib/linguist/heuristics.rb)
+* [vendor.yml](https://github.com/github/linguist/blob/master/lib/linguist/vendor.yml)
+* [documentation.yml](https://github.com/github/linguist/blob/master/lib/linguist/documentation.yml)
 
-*enry* re-uses parts of original [linguist](https://github.com/github/linguist) especially data in `languages.yml` to generate internal data structures. In oreder to update to latest upstream run
+For the moment we don't have any procedure established to detect changes in the linguist project automatically and regenerate the code. So we are updating the generated code when is needed  without any specific criteria while we are going ahead in the project development.
```
Comment (abeaumont) [inline README.md]: * s/generated code when is needed  without/generated code as needed, without/
* The last part (_while we are going...) not specially informative, could be removed
